nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/security/impl/ACPImpl.java

```diff
@@ -34,18 +34,29 @@
 import org.nuxeo.ecm.core.api.security.Access;
 import org.nuxeo.ecm.core.api.security.SecurityConstants;
 import org.nuxeo.ecm.core.api.security.UserEntry;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.services.config.ConfigurationService;
 
 /**
  * The ACP implementation uses a cache used when calling getAccess().
  */
 public class ACPImpl implements ACP {
 
+    /**
+     * ConfigurationService property to enable legacy behavior.
+     *
+     * @since 10.2
+     */
+    public static String LEGACY_BEHAVIOR_PROPERTY = "nuxeo.security.acl.legacyBehavior";
+
     private static final long serialVersionUID = 1L;
```

Comment:
Good point, I've always use 1L, and sometimes (maybe once over 1M) I incremented it to 2L.
What's the benefit to have a generated one ?

Reply:
Since I couldn't explain it well, here's a good link https://www.mkyong.com/java-best-practices/understand-the-serialversionuid/
But it does not give a better a reason why a generated value would be better than 1L.
I guess it's thought to avoid cases where different classes would a compatible signature.